deploy/multideploy.sh

```diff
@@ -0,0 +1,274 @@
+    _err "Failed to preprocess deploy file."
+    return 1
+  fi
+  _debug3 "File" "$file"
+
+  # Deploy to services
+  _deploy_services "$file"
+  _exitCode="$?"
+
+  # Save deployhook for renewals
+  _debug2 "Setting Le_DeployHook"
+  _savedomainconf "Le_DeployHook" "multideploy"
+
+  return "$_exitCode"
+}
+
+# Description:
+#   This function preprocesses the deploy file by checking if 'yq' is installed,
+#   verifying the existence of the deploy file, and ensuring only one deploy file is present.
+# Arguments:
+#   $@ - Posible deploy file names.
+# Usage:
+#   _preprocess_deployfile "<deploy_file1>" "<deploy_file2>?"
+_preprocess_deployfile() {
+  # Check if yq is installed
+  if ! command -v yq >/dev/null 2>&1; then
+    _err "yq is not installed! Please install yq and try again."
+    return 1
+  fi
+  _debug3 "yq is installed."
+
+  # Check if deploy file exists
+  IFS=$(printf '\n')
+  for file in "$@"; do
+    _debug3 "Checking file" "$DOMAIN_PATH/$file"
+    if [ -f "$DOMAIN_PATH/$file" ]; then
+      _debug3 "File found"
+      if [ -n "$found_file" ]; then
+        _err "Multiple deploy files found. Please keep only one deploy file."
+        return 1
+      fi
+      found_file="$file"
+    else
+      _debug3 "File not found"
+    fi
+  done
+  IFS=$OLDIFS
+
+  if [ -n "$found_file" ]; then
+    _check_deployfile "$DOMAIN_PATH/$found_file"
+  else
+    _err "Deploy file not found. Go to https://github.com/acmesh-official/acme.sh/wiki/deployhooks#36-deploying-to-multiple-services-with-the-same-hooks to see how to create one."
+    return 1
+  fi
+
+  echo "$DOMAIN_PATH/$found_file"
+}
+
+# Description:
+#   This function checks the deploy file for version compatibility and the existence of the specified configuration and services.
+# Arguments:
+#   $1 - The path to the deploy configuration file.
+#   $2 - The name of the deploy configuration to use.
+# Usage:
+#   _check_deployfile "<deploy_file_path>"
+_check_deployfile() {
+  _deploy_file="$1"
+  _debug2 "Deploy file" "$_deploy_file"
+
+  # Check version
+  _deploy_file_version=$(yq '.version' "$_deploy_file")
+  if [ "$MULTIDEPLOY_VERSION" != "$_deploy_file_version" ]; then
+    _err "As of $PROJECT_NAME $VER, the deploy file needs version $MULTIDEPLOY_VERSION! Your current deploy file is of version $_deploy_file_version."
+    return 1
+  fi
+  _debug2 "Deploy file version is compatible: $_deploy_file_version"
+
+  # Extract all services from config
+  _services=$(yq e '.services[].name' "$_deploy_file")
+  _debug2 "Services" "$_services"
+
+  if [ -z "$_services" ]; then
+    _err "Config does not have any services to deploy to."
+    return 1
+  fi
+  _debug2 "Config has services."
+
+  IFS=$(printf '\n')
+  # Check if extracted services exist in services list
+  for _service in $_services; do
+    _debug2 "Checking service" "$_service"
+    # Check if service exists
+    if ! yq e ".services[] | select(.name == \"$_service\")" "$_deploy_file" >/dev/null; then
```

Comment:
This check assumes that yq will exit with a non-zero state when it cannot find the selected service, but in my tests with the project's docker image, it exists with a zero every time, so the checks succeed even when the services are invalid.

Reply:
This is fixed in my branch